### PR TITLE
Add feature for project edit/update

### LINKED ISSUE #35: [Feature] [FE + BE] Individual Projects Edit and Update
Currently, our projects page look like this:
Image

1. Add the edit icon for the Project card according to the design.
Image

2. Create functionality for the Edit and Update feature for the projects. Create a form similar to the New Project Form but call it "Edit/Update Project". Only show IRB Number for admin. Ensure that the Edits get posted into our Database and is shown on the project live.

Image

Figma Link: https://www.figma.com/team_invite/redeem/m7USct7pyXUSdm1R6BbS9B

### PR DESCRIPTION
**Pull request by Will15800**
Added the edit icon for the Project card. Implemented functionality for the Edit and Update feature for the projects. Created a form to edit and update individual projects. The changes are updated in the database too. If the user is admin(based on user_type in User table) the IRB will show

### CODE REVIEW

File: src/components/EditProject.jsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+        if (!error) {
+          setUserData(user);
+  
+          const { data, error } = await supabase
+            .from("Users")
+            .select("user_type")
+            .eq("user_id", user.id);
+            
+          if (!error && data.length > 0 && data[0].user_type) {
```
Comment (RichardSima2021) [inline src/components/EditProject.jsx]: Encapsulate logic into helper function for code cleanup

**Comment by Will15800** [comment]
I did some code cleanup and created helper functions, also resolved merge conflict

**Comment by jessieh9** [comment]
> I did some code cleanup and created helper functions, also resolved merge conflict

I see, in the future you can leave merge conflicts to us so it's less work for the devs :))

File: src/components/ProjectCard.jsx [hunk 2]
```diff
@@ -84,9 +85,13 @@ function ProjectCard({ project }) {
                 {formatDate(project.created_at)}
               </Text>
             </Text>
+            <Spacer />
```
Comment (jessieh9) [inline src/components/ProjectCard.jsx:80]: Is this imported from Chakra?

File: src/components/ProjectCard.jsx [hunk 3]
```diff
@@ -84,9 +85,13 @@ function ProjectCard({ project }) {
...
+            <EditProject
```
Comment (jessieh9) [inline src/components/ProjectCard.jsx:81]: If you use this, remove the Pencil Icon, we don't need two edit buttons.

**Review by jessieh9 — CHANGES_REQUESTED** [review]

File: src/components/ProjectCard.jsx (hunk 3 above)
Comment (Will15800) [reply src/components/ProjectCard.jsx:81]: Yes I am using another Icon inside EditProject. I will remove the pencil icon

File: src/components/ProjectCard.jsx (hunk 2 above)
Comment (Will15800) [reply src/components/ProjectCard.jsx:80]: Yes it's used to create space between Edit and Project date

Comment (jessieh9) [reply src/components/ProjectCard.jsx:80]: Can you import it? It's not imported yet.

Comment (Will15800) [reply src/components/ProjectCard.jsx:80]: Got it

**Review by jessieh9 — APPROVED** [review]
LGTM

**Pull request merged**